File: frontend/src/types/proposal.ts
export type ProposalStatus =
  | 'draft'
  | 'discussion'
  | 'awaiting_review'
  | 'threshold_reached'
  | 'voting'
  | 'approved'
  | 'rejected'
  | 'withdrawn'

export type ProposalType =
  | 'amendment'
  | 'new_law'
  | 'repeal'
  | 'budget_alteration'

export interface Proposal {
  id: number
  number: string
  slug: string
  repository_id: number
  author_id: number
  title: string
  summary: string
  justification: string
  full_text: string
  type: ProposalType
  status: ProposalStatus
  branch_name: string
  target_branch: string
  signatures_count: number
  comments_count: number
  votes_count: number
  quorum_required: number | null
  threshold_percentage: number | null
  created_at: string
  updated_at: string
}

File: backend/app/schemas/proposal.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.models.proposal import ProposalStatus, ProposalType


class ProposalBase(BaseModel):
    title: str
    summary: str
    justification: str
    full_text: str
    type: ProposalType = Field(default=ProposalType.AMENDMENT)
    branch_name: str = Field(default="feature/nova-proposta")
    target_branch: str = Field(default="main")


class ProposalCreate(ProposalBase):
    pass


class ProposalUpdate(BaseModel):
    title: Optional[str] = None
    summary: Optional[str] = None
    justification: Optional[str] = None
    full_text: Optional[str] = None
    type: Optional[ProposalType] = None
    status: Optional[ProposalStatus] = None
    branch_name: Optional[str] = None
    target_branch: Optional[str] = None


class Proposal(ProposalBase):
    id: int
    number: str
    slug: str
    status: ProposalStatus
    repository_id: int
    author_id: int
    signatures_count: int
    comments_count: int
    votes_count: int
    quorum_required: Optional[int] = None
    threshold_percentage: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
